narwhals/translate.py

```diff
@@ -108,27 +121,30 @@ def from_native(
     eager_only: None = ...,
     series_only: None = ...,
     allow_series: None = ...,
+    api_version: str | None = ...,
 ) -> DataFrame | LazyFrame: ...
 
 
 @overload
 def from_native(
     native_dataframe: Any,
     *,
-    strict: bool,
-    eager_only: bool | None,
-    series_only: bool | None,
-    allow_series: bool | None,
+    strict: bool = ...,
+    eager_only: bool | None = ...,
+    series_only: bool | None = ...,
+    allow_series: bool | None = ...,
+    api_version: str | None = ...,
 ) -> DataFrame | LazyFrame | Series: ...
 
 
-def from_native(
+def from_native(  # noqa: PLR0913
     native_dataframe: Any,
     *,
     strict: bool = True,
     eager_only: bool | None = None,
     series_only: bool | None = None,
     allow_series: bool | None = None,
+    api_version: str | None = None,
```

Comment:
I know it is still WIP but commenting just in case:

- The argument is not documented in the docstring
- Is there a reason to be `None` instead of the default which is used otherwise?
- Maybe most important: I am not expecting to support any single version, I think we should limit and validate the value passed through.
 
Example combining the latter two points:

```py
Versions = Literal["0.20", "1.0"]
...
    
def from_native(
    ...
    api_version: Versions  = "0.20",
):

    if api_version not in get_args(Versions): Raise(...)
```

Reply:
yup, thanks! this is just a start, but it works well enough that the scikit-lego / timebased-cv tests suites pass

I'll post the diff in the linked issue to solicit feedback